operators/src/machine_learning/mod.rs

```diff
@@ -18,12 +21,23 @@ pub enum MachineLearningError {
         "Onnx model must have two dimensional input ([-1, b], b > 0). Found [{}].",
         dimensions.iter().map(std::string::ToString::to_string).collect::<Vec<_>>().join(", ")
     ))]
-    InvalidInputDimensions { dimensions: Vec<i64> },
+    InvalidDimensions { dimensions: Vec<i64> },
     #[snafu(display(
-        "Onnx model must have one dimensional output. Found [{}].",
-        dimensions.iter().map(std::string::ToString::to_string).collect::<Vec<_>>().join(", ")
+        "Onnx model must have one dimensional output or match the tiling tile size. Found {:?} and {:?}.",
+        tensor_shape, tiling_shape
     ))]
-    InvalidOutputDimensions { dimensions: Vec<i64> },
+    InvalidInputShape {
+        tensor_shape: TensorShape3D,
+        tiling_shape: GridShape2D,
+    },
+    #[snafu(display(
+        "Onnx model must have one dimensional output or match the tiling tile size. Found {:?} and {:?}.",
```

Comment:
just a question: wouldn't it also be relatively easy to support model outputs that are multiples/divided by two?

Reply:
yes but then the input might get tricky. There is still a lot to do here but maybe not in this PR